shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Protocol.hs

```diff
@@ -91,6 +91,7 @@ mkPrtclEnv ::
   -- | New epoch marker. This should be true iff this execution of the PRTCL
   -- rule is being run on the first block in a new epoch.
   Bool ->
+  Nonce ->
```

Comment:
@mrBliss Is it okay that I've added the prev-hash-nonce to this API function which creates the Prtcl environment? This value is not a part of the new epoch state (ie ledger view).

Reply:
From the ledger point of view this is 100% the right thing to do, I think.

~For consensus it means slightly more work, because now we have to store this `Nonce` on our side in the `ConsensusState` so that we can pass it to `mkPrtclEnv`, previously, it was just in the `PrtclState` so we didn't have to store it manually.~

Oh, but I just realised that where we call `mkPrtclEnv`, we have the whole block header in scope, which also contains the "previous hash", so we can use that to construct the nonce. No need to store it in the state. Great, triple win (ledger + consensus + less memory usage: we have to store snapshots of the past `k` `PrtclState` to allow rewinding when switching to a fork, so making `PrtclState` smaller is a win)!

Reply:
great! :)